File: app/javascript/src/hooks/get_currencies.ts
import { JobSalaryCurrency } from '@customtypes/job_post';
import { useState, useEffect } from 'react';

export const getCurrencies = () => {
    const [currencies, setCurrencies] = useState<JobSalaryCurrency[]>([]);
    const [loading, setLoading] = useState<boolean>(true);
    const [error, setError] = useState<string | null>(null);

    useEffect(() => {
        const fetchCurrencies = async () => {
            try {
                setLoading(true);
                setError(null);

                const response = await fetch('/job_salary_currencies.json');
                if (!response.ok) {
                    throw new Error(`Error fetching currencies: ${response.status}`);
                }

                const data = await response.json();
                setCurrencies(data);
            } catch (err) {
                setError(
                    err instanceof Error ? err.message : 'An unknown error occurred'
                );
            } finally {
                setLoading(false);
            }
        };

        fetchCurrencies();
    }, []);

    return { currencies, loading, error };
}

File: app/python/hooks/get_currencies.py
#  app/python/hooks/get_currencies.py
import requests


def fetch_currencies():
    try:
        response = requests.get("http://localhost:3000/job_salary_currencies.json")
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error fetching currencies: {e}")
        return None


#  cache data
currency_data = fetch_currencies()


def get_currency_type(symbol):
    """Fetch currency code from cached currency data based on symbol."""
    if not currency_data:
        return None
    for currency in currency_data:
        if currency["currency_sign"] == symbol:
            return currency["currency_code"]
    return None
